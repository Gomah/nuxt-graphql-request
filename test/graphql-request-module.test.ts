import { fileURLToPath } from 'node:url';
import { describe, expect, test } from 'vitest';
import { setup, $fetch } from '@nuxt/test-utils';
import { load as cheerio } from 'cheerio';
import htmlSerializer from 'jest-serializer-html';

expect.addSnapshotSerializer(htmlSerializer);

describe('SSR', async () => {
  await setup({
    rootDir: fileURLToPath(new URL('./fixture', import.meta.url)),
  });

  test('Nuxt GraphQL Request', async () => {
    const html = await $fetch('/');
    const $ = cheerio(html);
    expect($('div#test').html()).toMatchSnapshot();
  });

  test('With AST', async () => {
    const html = await $fetch('/with-ast');
    const $ = cheerio(html);
    expect($('div#test').html()).toMatchSnapshot();
  });

  test('Using batch queries', async () => {
    const html = await $fetch('/batch');
    const $ = cheerio(html);
    expect($('div#test').html()).toMatchSnapshot();
  });

  test('With .gql Import', async () => {
    const html = await $fetch('/import');
    const $ = cheerio(html);
    expect($('div#test').html()).toMatchSnapshot();
  });
});
