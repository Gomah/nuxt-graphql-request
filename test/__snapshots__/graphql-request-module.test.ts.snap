// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SSR > Nuxt GraphQL Request 1`] = `
<h2>
  Countries
</h2>
<ul>
  <!--[-->
  <li>
    Andorra
  </li>
  <li>
    Austria
  </li>
  <li>
    Åland
  </li>
  <li>
    Belgium
  </li>
  <li>
    Saint Barthélemy
  </li>
  <li>
    Cyprus
  </li>
  <li>
    Germany
  </li>
  <li>
    Estonia
  </li>
  <li>
    Spain
  </li>
  <li>
    Finland
  </li>
  <li>
    France
  </li>
  <li>
    French Guiana
  </li>
  <li>
    Guadeloupe
  </li>
  <li>
    Greece
  </li>
  <li>
    Ireland
  </li>
  <li>
    Italy
  </li>
  <li>
    Lithuania
  </li>
  <li>
    Luxembourg
  </li>
  <li>
    Latvia
  </li>
  <li>
    Monaco
  </li>
  <li>
    Montenegro
  </li>
  <li>
    Saint Martin
  </li>
  <li>
    Martinique
  </li>
  <li>
    Malta
  </li>
  <li>
    Netherlands
  </li>
  <li>
    Saint Pierre and Miquelon
  </li>
  <li>
    Portugal
  </li>
  <li>
    Réunion
  </li>
  <li>
    Slovenia
  </li>
  <li>
    Slovakia
  </li>
  <li>
    San Marino
  </li>
  <li>
    French Southern Territories
  </li>
  <li>
    Vatican City
  </li>
  <li>
    Kosovo
  </li>
  <li>
    Mayotte
  </li>
</ul>
<h2>
  Star wars movies
</h2>
<ul>
  <!--[-->
  <li>
    A New Hope
  </li>
  <li>
    The Empire Strikes Back
  </li>
  <li>
    Return of the Jedi
  </li>
  <li>
    The Phantom Menace
  </li>
  <li>
    Attack of the Clones
  </li>
  <li>
    Revenge of the Sith
  </li>
</ul>
`;

exports[`SSR > Using batch queries 1`] = `
<h2>
  Countries using EUR
</h2>
<ul>
  <!--[-->
  <li>
    Andorra
  </li>
  <li>
    Austria
  </li>
  <li>
    Åland
  </li>
  <li>
    Belgium
  </li>
  <li>
    Saint Barthélemy
  </li>
  <li>
    Cyprus
  </li>
  <li>
    Germany
  </li>
  <li>
    Estonia
  </li>
  <li>
    Spain
  </li>
  <li>
    Finland
  </li>
  <li>
    France
  </li>
  <li>
    French Guiana
  </li>
  <li>
    Guadeloupe
  </li>
  <li>
    Greece
  </li>
  <li>
    Ireland
  </li>
  <li>
    Italy
  </li>
  <li>
    Lithuania
  </li>
  <li>
    Luxembourg
  </li>
  <li>
    Latvia
  </li>
  <li>
    Monaco
  </li>
  <li>
    Montenegro
  </li>
  <li>
    Saint Martin
  </li>
  <li>
    Martinique
  </li>
  <li>
    Malta
  </li>
  <li>
    Netherlands
  </li>
  <li>
    Saint Pierre and Miquelon
  </li>
  <li>
    Portugal
  </li>
  <li>
    Réunion
  </li>
  <li>
    Slovenia
  </li>
  <li>
    Slovakia
  </li>
  <li>
    San Marino
  </li>
  <li>
    French Southern Territories
  </li>
  <li>
    Vatican City
  </li>
  <li>
    Kosovo
  </li>
  <li>
    Mayotte
  </li>
</ul>
<h2>
  Countries using AUD
</h2>
<ul>
  <!--[-->
  <li>
    Australia
  </li>
  <li>
    Cocos [Keeling] Islands
  </li>
  <li>
    Christmas Island
  </li>
  <li>
    Heard Island and McDonald Islands
  </li>
  <li>
    Kiribati
  </li>
  <li>
    Norfolk Island
  </li>
  <li>
    Nauru
  </li>
  <li>
    Tuvalu
  </li>
</ul>
`;

exports[`SSR > With .gql Import 1`] = `
<h2>
  Countries
</h2>
<ul>
  <!--[-->
  <li>
    American Samoa
  </li>
  <li>
    Bonaire
  </li>
  <li>
    Ecuador
  </li>
  <li>
    Micronesia
  </li>
  <li>
    Guam
  </li>
  <li>
    British Indian Ocean Territory
  </li>
  <li>
    Marshall Islands
  </li>
  <li>
    Northern Mariana Islands
  </li>
  <li>
    Puerto Rico
  </li>
  <li>
    Palau
  </li>
  <li>
    Turks and Caicos Islands
  </li>
  <li>
    East Timor
  </li>
  <li>
    U.S. Minor Outlying Islands
  </li>
  <li>
    British Virgin Islands
  </li>
  <li>
    U.S. Virgin Islands
  </li>
</ul>
<h2>
  Star wars movies
</h2>
<ul>
  <!--[-->
  <li>
    A New Hope
  </li>
  <li>
    The Empire Strikes Back
  </li>
  <li>
    Return of the Jedi
  </li>
  <li>
    The Phantom Menace
  </li>
  <li>
    Attack of the Clones
  </li>
  <li>
    Revenge of the Sith
  </li>
</ul>
`;

exports[`SSR > With AST 1`] = `
<h2>
  Countries
</h2>
<ul>
  <!--[-->
  <li>
    United Kingdom
  </li>
  <li>
    Guernsey
  </li>
  <li>
    South Georgia and the South Sandwich Islands
  </li>
  <li>
    Isle of Man
  </li>
  <li>
    Jersey
  </li>
</ul>
<h2>
  Star wars movies
</h2>
<ul>
  <!--[-->
  <li>
    A New Hope
  </li>
  <li>
    The Empire Strikes Back
  </li>
  <li>
    Return of the Jedi
  </li>
  <li>
    The Phantom Menace
  </li>
  <li>
    Attack of the Clones
  </li>
  <li>
    Revenge of the Sith
  </li>
</ul>
`;
